### DISCUSSION TITLE
Web container created with service name wtb instead of web?

### DISCUSSION THREAD

**Discussion by deviantintegral** [ddev/ddev#5470] [Q&A] [answered]
In the last few days, we've started to have an odd error in our GitHub actions running ddev. It's not consistent. For example, we have a matrix build to run automated tests, and all of the code to this point is the same. Yet, some of the jobs will get past the part fine, while others will fail.

The key bit is the container name. Note it's `ddev-redacted-wtb` not `ddev-redacted-web`:

```
2023-10-28T13:19:32.3543568Z Building project images...
2023-10-28T13:20:25.6834691Z [32mProject images built in 53s.[0m
2023-10-28T13:20:25.7867556Z  Network ddev-redacted_default  Created
2023-10-28T13:20:25.7882506Z  Volume "ddev-redacted_sqlite_database"  Created
2023-10-28T13:20:25.7892154Z  Volume "ddev-redacted_database"  Created
2023-10-28T13:20:25.8056685Z  Container ddev-redacted-redis  Created
2023-10-28T13:20:25.8062742Z  Container ddev-redacted-wtb  Created
2023-10-28T13:20:25.8063619Z  Container ddev-redacted-db  Created
2023-10-28T13:20:26.1975666Z  Container ddev-redacted-redis  Started
2023-10-28T13:20:26.3890056Z  Container ddev-redacted-wtb  Started
2023-10-28T13:20:26.5696937Z  Container ddev-redacted-db  Started
2023-10-28T13:20:28.2468906Z .........................[31mFailed to start redacted: composer.json not found in container, stdout='', stderr='': service web does not exist in project redacted (state=doesnotexist); please report this situation, https://github.com/ddev/ddev/issues; probably can be fixed with ddev restart[0m
2023-10-28T13:20:28.2502647Z ##[error]Process completed with exit code 1.
```

Our first thought was that this was somehow related to the 1.22.4 release as the timing lines up, but no one has seen this happen on a local.

I searched the ddev code for any possible typos of `wtb` and didn't find any. Has anyone else seen this come up? I think our next step will be see about downgrading ddev (pending https://github.com/ddev/github-action-setup-ddev/issues/13) to determine if the issue goes away. My other thought was that this could somehow be a bug in a new Docker CE release, but I don't know of any reasonable way for us to test that in Actions.

## Error summary

* `Failed to start redacted: composer.json not found in container, stdout='', stderr='': service web does not exist in project redacted (state=doesnotexist)` (container service is called `-wtb`)
* `Failed to restart REDACTED: docker-compose build failed: composeCmd failed to run 'COMPOSE_PROJECT_NAME=ddev-REDACTED docker-compose -f /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml --progress=plain build', action='[]', err='exit status 15', stdout='', stderr='validating /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml: services.web Additional property environtent is not allowed', output='', stderr='validating /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml: services.web Additional property environtent is not allowed` (`environment` is `environtent`)
* `Failed to start REDACTED: yaml: line 126: did not find expected key` (line 126 in `.ddev/**/*y?ml` either doesn't exist, is an empty line, or is a valid line of yaml)

**Comment by rfay** [comment]
This is pretty concerning.

I think the way to test it is by running the test manually using tmate and seeing if you can recreate it there. 

I've seen some weird results in the nightly build on https://github.com/ddev/ddev-platformsh recently, but they seem to be intermittent and I havent chased them. But at least one of them looked like data corruption, but IIRC it wasn't anything like what you're seeing here. 

Thanks for raising it, happy to help you look, we can do a screenshare and do tmate. If you don't have tmate set up, you'll want it. All the add-ons have it by default and most of ddev's tests have it.

**Reply by deviantintegral** [reply]
Notes as I investigate:

```
$ grep -r wtb .ddev
.ddev/.ddev-docker-compose-full.yaml:        container_name: ddev-REDACTED-wtb

$ ddev debug dockercheck
Docker platform: Ubuntu 22.04.3 LTS
Using Docker context: default (unix:///var/run/docker.sock)
docker-compose: v2.21.0
Using DOCKER_HOST=unix:///var/run/docker.sock
Docker version: 24.0.6
Able to run simple container that mounts a volume.
Able to use internet inside container.

$ ddev start
Starting redacted...
No output is shown during the first container build. It may take a minute to install CLI tools.
Using custom PHP configuration: [/home/runner/work/redacted/redacted/.ddev/php/upload-limit.ini]
Custom configuration is updated on restart.
If you don't see your custom configuration taking effect, run 'ddev restart'.
Building project images...
Project images built in 0s.
 Container ddev-redacted-redis  Running
 Container ddev-redacted-db  Recreate
 Container ddev-redacted-wtb  Recreate
 Container ddev-redacted-db  Recreated
 Container ddev-redacted-wtb  Recreated
 Container ddev-redacted-db  Started
 Container ddev-redacted-web  Started
Waiting for web/db containers to become ready: [web db]
Starting web_extra_daemons...
```

That's surprising that rerunning ddev start fixed it up. 

I've run ` ddev delete -Oy && git clean -dxf .ddev && ddev start` a few times and it hasn't reproduced.

Running `ddev debug compose-config` also fixes this, correcting the `.ddev/.ddev-docker-compose-full.yaml` file.

**Reply by rfay** [reply]
The logic for ddev is all in https://github.com/ddev/ddev/blob/989c1732dcc1467fc568a222c3e9069fd3a2a1ed/pkg/ddevapp/ddevapp.go#L1220-L1283

So the time for data corruption to happen (.ddev-docker-compose-full.yaml being broken?) is in that area. 

WriteDockerComposeYaml, where it could happen in the first place, is https://github.com/ddev/ddev/blob/b1810f3ad63de4036b6df1f7f6c1f777a11e4201/pkg/ddevapp/compose_yaml.go#L17-L72

fixupComposeYaml() seems like a suspicious place, since we fiddle with it there:

https://github.com/ddev/ddev/blob/b1810f3ad63de4036b6df1f7f6c1f777a11e4201/pkg/ddevapp/compose_yaml.go#L76-L124

**Reply by deviantintegral** [reply]
I've now run 24 CI jobs in GitHub with ddev 1.22.3, and none of them failed. So, that's got me thinking that this is a change in 1.22.4 that's causing this.

**Reply by rfay** [reply]
It's intermittent, true?

**Reply by deviantintegral** [reply]
Yeah, I would say 2/6 of jobs run will fail with 1.22.4.

What surprises me the most is that it's the exact same issue each time. Like why the character `t` and not something else, or something different each time? Like if it was a bad CI node that we kept hitting, I'd expect more randomness.

**Reply by rfay** [reply]
Sometimes there is a difference in runner versions, and that causes very complex things.

**Reply by rfay** [reply]
Will be looking forward to solving this. I think tmate may be the way to do it, and also carefully inspecting the runs to make sure we know what version of runner was running. I don't know how the .ddev/.docker-compose-base.yaml or .ddev-docker-compose-full.yaml can be corrupted, assuming that's what's happening.

Oh, it could easily happen if DDEV_SITENAME were somehow corrupted. .ddev/.ddev-docker-compose-base.yaml has this:
```
web:
    container_name: ddev-${DDEV_SITENAME}-web
```

Is there anything anywhere in your stuff that touches the environment variable DDEV_SITENAME?

**Reply by deviantintegral** [reply]
> Sometimes there is a difference in runner versions, and that causes very complex things.

Runner versions are the same between passing and failing jobs. For example:

```
2023-10-31T16:42:10.4906721Z Requested labels: ubuntu-22.04-l
2023-10-31T16:42:10.4906926Z Job defined at: REDACTED/REDACTED/.github/workflows/Playwright.yml@refs/pull/1874/merge
2023-10-31T16:42:10.4907056Z Waiting for a runner to pick up this job...
2023-10-31T16:42:16.6688746Z Job is waiting for a runner from 'ubuntu-22.04-l' to come online.
2023-10-31T16:42:16.8297269Z Job is about to start running on the runner: ubuntu-22.04-l_3ec0c9488ed1 (organization)
2023-10-31T16:42:20.7705949Z Current runner version: '2.309.0'
2023-10-31T16:42:20.7712118Z Runner name: 'ubuntu-22.04-l_3ec0c9488ed1'
2023-10-31T16:42:20.7712890Z Runner group name: 'Default'
2023-10-31T16:42:20.7713652Z Machine name: 'runner'
2023-10-31T16:42:20.7729618Z ##[group]Operating System
2023-10-31T16:42:20.7730206Z Ubuntu
2023-10-31T16:42:20.7730714Z 22.04.3
2023-10-31T16:42:20.7731086Z LTS
2023-10-31T16:42:20.7731706Z ##[endgroup]
2023-10-31T16:42:20.7732126Z ##[group]Runner Image
2023-10-31T16:42:20.7732610Z Image: ubuntu-22.04
2023-10-31T16:42:20.7733101Z Version: 20231025.1.0
2023-10-31T16:42:20.7734146Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20231025.1/images/linux/Ubuntu2204-Readme.md
2023-10-31T16:42:20.7735707Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20231025.1
2023-10-31T16:42:20.7736611Z ##[endgroup]
2023-10-31T16:42:20.7738837Z ##[group]GITHUB_TOKEN Permissions
2023-10-31T16:42:20.7740771Z Actions: write
2023-10-31T16:42:20.7741490Z Checks: write
2023-10-31T16:42:20.7741996Z Contents: write
2023-10-31T16:42:20.7742480Z Deployments: write
2023-10-31T16:42:20.7742916Z Discussions: write
2023-10-31T16:42:20.7743330Z Issues: write
2023-10-31T16:42:20.7743796Z Metadata: read
2023-10-31T16:42:20.7744202Z Packages: write
2023-10-31T16:42:20.7744741Z Pages: write
2023-10-31T16:42:20.7745252Z PullRequests: write
2023-10-31T16:42:20.7745715Z RepositoryProjects: write
2023-10-31T16:42:20.7746266Z SecurityEvents: write
2023-10-31T16:42:20.7746702Z Statuses: write
2023-10-31T16:42:20.7747118Z ##[endgroup]
2023-10-31T16:42:20.7749258Z Secret source: Actions
2023-10-31T16:42:20.7749883Z Prepare workflow directory
2023-10-31T16:42:20.8365308Z Prepare all required actions
2023-10-31T16:42:20.8522604Z Getting action download info
2023-10-31T16:42:21.0861554Z Download action repository 'actions/checkout@v3' (SHA:f43a0e5ff2bd294095638e18286ca9a3d1956744)
2023-10-31T16:42:21.4732211Z Download action repository 'actions/cache@v3' (SHA:704facf57e6136b1bc63b828d79edcd491f0ee84)
2023-10-31T16:42:21.8427371Z Download action repository 'actions/upload-artifact@v3' (SHA:a8a3f3ad30e3422c9c7b888a15615d19a852ae32)
2023-10-31T16:42:22.2585531Z Complete job name: Playwright (4)
```

`3ec0c9488ed1` seems to change each run so I don't think we're hitting a bad CI node.

> Is there anything anywhere in your stuff that touches the environment variable DDEV_SITENAME?

We only read the sitename, never set it.

> tmate may be the way to do it

Yes, that's how I found the broken compose file above. I also added `export DDEV_DEBUG=true` but nothing useful in those additional logs so far.

Since we can now pin the version in CI, we may wait further to investigate on the hopes that someone else sees the same issue outside of GitHub actions. Or, perhaps we could roll a custom ddev build that logs if it ever sees this comes up and logs helpful debug info.

**Reply by rfay** [reply]
In tmate, are you able to `ddev restart` multiple times without failure? I would think that would show this. It may be worth examining environment variables in there, `env | grep DDEV_`

I didn't see that you had given a .ddev-docker-compose-full.yaml or .ddev-docker-compose-base.yaml previously, but would you be willing to PM those to me if you can capture them?

**Reply by deviantintegral** [reply]
Yeah, I was able to restart several times and it never failed. But perhaps I didn't restart enough.

I'll grab a copy of the compose files (but later today or tomorrow).

I updated a PR that tested pulling in 1.22.4 again, and it failed for me. Since I'm now confident that this must somehow be a regression in 1.22.3 (or site-specific code that's only broken in 1.22.4), I'm going to try bisecting commits in CI. I'm not going to fully automate it (I'll manually commit the ddev binary to test), but it's only estimating 5 steps so if we're lucky we'll have a bad commit to look at in the next day or so.

**Reply by rfay** [reply]
Try `set -eu` and then `while true; do ddev restart; done` and it will fail when it fails. It might give us a clue. 

Bisecting commits is a great technique too.

**Reply by deviantintegral** [reply]
In the process of bisecting, `environment` became `environtent`! This is on commit `c6a8acd6762d68626fef95e408413865f76545b5`, bisecting is still ongoing.

```
2023-11-01T12:02:01.4392100Z validating /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml: services.web Additional property environtent is not allowed
2023-11-01T12:02:01.4417273Z [31mFailed to restart REDACTED: docker-compose build failed: composeCmd failed to run 'COMPOSE_PROJECT_NAME=ddev-REDACTED docker-compose -f /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml --progress=plain build', action='[]', err='exit status 15', stdout='', stderr='validating /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml: services.web Additional property environtent is not allowed', output='', stderr='validating /home/runner/work/REDACTED/REDACTED/.ddev/.ddev-docker-compose-full.yaml: services.web Additional property environtent is not allowed'
```

**Reply by rfay** [reply]
I hope you'll track/document your full bisecting process and technique. I've used manual bisecting many times, but for something like this that is intermittent a better technique than manual is important.

**Reply by deviantintegral** [reply]
More oddities. Bisecting didn't leave anything conclusive - it said every commit was good except for the 1.22.4 commit. Multiple commits would pass and fail. I've now seen a third error:

```
Failed to start REDACTED: yaml: line 126: did not find expected key
```

I downloaded the .ddev directory from the failed build, and there's nothing wrong with line 126 on any yaml file. They all pass yaml-lint too.

I also had a while where I was consistently getting the above error. Then I created a new PR with the same contents... and I didn't, for a while at least.

This has all got me back to your suggestion that this is really an issue with runners. I'm leaning towards that as well. I am now seeing different runner image versions on different job runs (`20231025.1.0` vs `20231030.2.1`), but both have failures. That got me looking in the runner issues, and there's a few that give me pause:

https://github.com/actions/runner-images/issues/8659
https://github.com/actions/runner-images/issues/8685

I'm concerned to see things like "upgrading core libraries from a PPA" without lots of docs and info around it. I'm thinking this may not be related to the ddev version, but either the environment in which the ddev release was compiled from _or_ some change in the ubuntu-22.04 image. I'm going to try downgrading the ubuntu image to 20.04, while at the same time see about cross-compiling a new build of ddev 1.22.4 on my mac.

For reference, my bisecting procedure was:

1. From a local clone of ddev, run `git bisect ...` between the last two releases.
2. `git bisect skip` any commits with `skip ci` since those wouldn't have an artifact to download.
3. Manually browse to the github commit on that hash, such as `https://github.com/ddev/ddev/commit/c6a8acd6762d68626fef95e408413865f76545b5`.
4. Download the build from the associated CI run.
5. Commit the ddev binary to the repo (I'm never merging this branch, and will delete it so it won't blow up our clone size long-term).
6. Replaced the install ddev action call with:

```
   - name: Install ddev from commit
       shell: bash
       run: |
         sudo cp -v ddev-* /usr/local/bin/ddev
         sudo chmod +x /usr/local/bin/ddev
         sudo curl -JLO "https://dl.filippo.io/mkcert/latest?for=linux/amd64"
         sudo chmod +x mkcert-v*-linux-amd64
         sudo cp mkcert-v*-linux-amd64 /usr/local/bin/mkcert
         mkcert -install
```

I also blew out our matrix to 12 jobs, and dropped down from a large to a regular runner, and disabled actually running functional tests. I'm not sure there's a way to truly automate bisect this way, since github generally prevents recursive actions.

**Reply by deviantintegral** [reply]
@rfay Ubuntu 20.04 failed with the `did not find expected key` error as well.

But, my self-compiled version of ddev 1.22.4 has now passed 48 jobs without failure. I compiled on my Mac (I love how easy cross compiling go is, and how you made it even easier for ddev!), so there's huge differences in the build environment. The binary has debug symbols and the like too.

I just don't like that no one else has chimed in with similar issues. Perhaps other users are just rerunning their CI builds, since they do pass most of the time? But if the build environment was somehow broken, wouldn't that also affect regular linux users, or even macos and windows users? I do suppose we are running `ddev start` way more often in CI due to the number of PRs our project has combined with running tests in parallel shards. It might take a week or two for someone to notice that `ddev start` is occasionally broken.

I think it's now time to wait on this and see if anyone else reports issues. If they do, let's have them try a separately built copy of the ddev binary. If not, we will try to upgrade our project again whenever the next ddev release is out, and if this is an issue in GitHub's runners then hopefully its fixed by then.

**Reply by rfay** [reply]
What go version did you build with? (`go version`)

DDEV's own tests mostly do their own build, but all the add-ons, which run nightly tests, are using the upstream builds, so should have problems identical to yours. But most would only do a couple of restarts of stable ddev per build. You're not seeing this on the add-ons you maintain right?

**Reply by deviantintegral** [reply]
```
$ go version
go version go1.21.3 darwin/arm64
```

Looks like I had three failures on HEAD for the playwright addon over the weekend. Last few days have been good though. I was assuming there were changes in the amplitude text that were fixed? https://github.com/deviantintegral/ddev-playwright/actions/runs/6664802645/job/18113113431

Otherwise, that's it.

**Reply by rfay** [reply]
The failures on HEAD probably wouldn't be related to build issues with distributed ones, because they (if using standard technique) do a `brew install --HEAD` which builds locally.

As of v1.22.4, the build was done on github using
```
      - uses: actions/setup-go@v4
        with:
          go-version: '>=1.20'
```

Which could possibly have resulted in a variety of go versions since we're now at 1.21.3. 

The actual build was in https://github.com/ddev/ddev/actions/runs/6655110294

Linux was done in https://github.com/ddev/ddev/actions/runs/6655110294/job/18084619605

It used go 1.21.3

I did see the 
> Instrumentation is opted in, but AmplitudeAPIKey is not available. This usually means you have a locally-built ddev binary or one from a PR build. It's not an error. Please report it if you're using an official release build.

I did see the amplitude warning you were seeing in HEAD and fixed it in https://github.com/ddev/ddev/pull/5471 - I saw it show up in a few add-ons. However, all tests should be running with amplitude turned off, so this shouldn't show up in the first place. If you're using github-action-add-on-test, this went into https://github.com/ddev/github-action-add-on-test/releases/tag/v1.1.1

**Comment by deviantintegral** [comment]
Here's my locally compiled 1.22.4: [ddev.tar.gz](https://github.com/ddev/ddev/files/13313621/ddev.tar.gz)

**Comment by deviantintegral** [comment]
We upgraded back to 1.22.4 in CI because we wanted to omit the database container in some jobs and there was a bugfix for that. So far things have been working fine again, though we're now seeing the runner image is version `20231030.2.0` which is somehow in between the versions we had previously seen.

**Reply by rfay** [reply]
Thanks for chasing and keeping up with this. Sure makes you wonder. They are doing massive maintenance and refactoring on the runner images (all platforms) so I imagine we'll see things happening. They say it will take months to complete.

**Comment by deviantintegral** [comment]
We upgraded to 1.22.5 a few days ago and are now seeing the same class of bug, but a different string! It's also not consistent and IMO is somehow an issue with the runners.

```
stderr='failed to solve: ddev/ddev-webeerver:v1.22.5: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed'
```

Note `webeerver` instead of `webserver`.

**Reply by rfay** [reply]
So weird. We haven't seen anything like this on our GitHub tests. Remind me which runner you're using. We have mostly Ubuntu 22.04.

**Reply by deviantintegral** [reply]
All Ubuntu 22.04. I set up a runner on Linode which is Debian 12 + the Actions Runner, and so far the issue hasn't come up there.

**Reply by penyaskito** [reply]
Happened again on the gh hosted runner with ubuntu 22.04, but this time is `websesver`, not `webserver` or `webeerver`

```
', stderr='failed to solve: ddev/ddev-websesver:v1.22.5: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed'
```

Note `websesver `instead of `webserver` (different typo than previous).

**Reply by rfay** [reply]
All of you are using https://github.com/ddev/github-action-setup-ddev ? Could that be part of the equation here?

**Reply by deviantintegral** [reply]
We are - and we _aren't_ in our self-hosted runners because they don't get reset between jobs. We're also not using `autostart` so that's not running either.

There's only a few commands in https://github.com/ddev/github-action-setup-ddev/blob/main/lib/main.js so I would be shocked if it was the issue.

Also for others, @penyaskito is working on the same project as me so we can't treat this as a second independent confirmation of the issue.

**Comment by deviantintegral** [comment]
😱 we've now seen this happen in our Linode hosted runners. I guess I can stop being annoyed at GitHub 😅 .

> Failed to start PROJECT: docker-compose build failed: composeCmd failed to run 'COMPOSE_PROJECT_NAME=ddev-PROJECT docker-compose -f /home/runner/actions-runner/_work/PROJECT/PROJECT/.ddev/.ddev-docker-compose-full.yaml --progress=plain build', action='[]', err='exit status 17', stdout='', stderr='unable to prepare context: path "/home/runner/aciions-runner/_work/PROJECT/PROJECT/.ddev/.webimageBuild" not found', output='', stderr='unable to prepare context: path "/home/runner/aciions-runner/_work/PROJECT/PROJECT/.ddev/.webimageBuild" not found'

We're going to capture the .ddev directory as an artifact. While I can't be 100% sure though (because I'm not taking runners offline), it doesn't appear that the YAML file on disk is corrupt. Next steps I see are:

1. If we can confirm the docker compose file is fine, then this could be a bug in docker compose itself. Perhaps we can downgrade it and see if that helps.
2. If the compose file is broken, then I think https://github.com/ddev/ddev/blob/master/pkg/ddevapp/compose_yaml.go is a place to start. Perhaps there's a concurrency bug with the serialize / deserialize yaml dance? I think we could consider adding in some additional debug logs to each step of generating the compose file so we can see where it goes wrong.

**Reply by deviantintegral** [reply]
I've confirmed the docker compose file is _not_ OK. In this latest error contains:

```
    web:
        build:
            args:
                BASE_IMAGE: ddev/ddev-websesver:v1.22.5
```

**Comment by davereid** [comment]
Oooooh I have seen this happen randomly in GitHub Actions as well, it's always a one-letter difference in the service name and it won't start correctly.

**Reply by davereid** [reply]
To clarify: this is completely different project/repo that I was seeing these random startup failure.

**Comment by deviantintegral** [comment]
This just happened in a public PR, so at least we can now share _everything_.

Failed: https://github.com/Lullabot/drainpipe/actions/runs/7727308894/job/21065661055
Rebuild: https://github.com/Lullabot/drainpipe/actions/runs/7727308894/job/21079350697

**Reply by rfay** [reply]
Well, that's progress. 

@davereid I hope you'll provide an example for yours as well. 

Someday somebody is going to come up with a repro case. Thanks for staying on this!

**Comment by rfay** [comment]
Is it always the docker image name `ddev/ddev-webserver` that is corrupted?

Is it always the BASE_IMAGE that's broken in .ddev/.ddev-docker-compose-full.yaml?
```
    web:
        build:
            args:
                BASE_IMAGE: ddev/ddev-webserver:v1.22.6
```

It would be pretty easy to sprinkle some greps into your test to try to find out where it's getting corrupted. You've probably already done that. But just `grep "BASE_IMAGE: ddev/ddev-webserver" .ddev/.ddev-docker-compose-full.yaml` would be an interesting test. If not found, the corruption has already happened.

I also note that that line is created from `BASE_IMAGE: $DDEV_WEBIMAGE` in .ddev/.ddev-docker-compose-base.yaml. So any corruption of the environment variable DDEV_WEBIMAGE would break it all. 

DDEV_WEBIMAGE is set here: https://github.com/ddev/ddev/blob/94d77509a582d92b5d975be97e84c66e0ad0a6de/pkg/ddevapp/ddevapp.go#L2106


Although it's discouraged, and not even in docs any more I don't think, there is a `webimage` key in .ddev/config.yaml that can be used to override the webimage.

**Comment by deviantintegral** [comment, marked as answer]
I got to the bottom if this and it's a bug introduced in 1.22.4. PR up at https://github.com/ddev/ddev/pull/5893.

**Reply by rfay** [reply]
Amazing and congratulations!
